samples/Basic_UPnP/README.rst

```diff
@@ -11,10 +11,6 @@ Services
 UPnP devices may also provide services which can be enumerated and used to control it.
 
 The Wemo device :cpp:class:`Wemo::Controllee` has two services for events and metadata.
-At present only the events interface is implemented.
-
-The service descriptions are stored in the ``wemo-service.xml`` file.
-The UPnP stack handles decoding and parsing of incoming requests, which then calls
-the :cpp:func:`UPnP::Service::handleAction` method which is overridden here by
-:cpp:class:`Wemo::BasicEventService`.
 
+The device and service descriptions are stored in the ``schema`` directory.
+This is used by UPnP to generate class code, so all we need to do is implement the action methods.
```

Comment:
I am really interested to see how this can be done. I am planning to create a sample [Sming UPnP application for an existing Sonoff TH16 switch](https://github.com/slaff/Sming/issues/64) at home.

Reply:
https://www.meshcommander.com/upnptools contains a `Service Author` program which can help with creating custom schema. It's Windows-only...